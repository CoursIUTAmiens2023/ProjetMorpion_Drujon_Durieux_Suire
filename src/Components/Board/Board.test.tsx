import { describe, expect, it } from "@jest/globals"
import { render } from "@testing-library/react"
import { Board } from "."

describe("Board", () => {
    it("should match the snapshot when square are null", () => {
        const { container } = render(
            <Board squares={Array(9).fill(null)} onClick={() => {}} />,
        )
        expect(container).toMatchSnapshot()
    })
    it("should match the snapshot when square are X", () => {
        const { container } = render(
            <Board squares={Array(9).fill("X")} onClick={() => {}} />,
        )
        expect(container).toMatchSnapshot()
    })
    it("should match the snapshot when square are O", () => {
        const { container } = render(
            <Board squares={Array(9).fill("O")} onClick={() => {}} />,
        )
        expect(container).toMatchSnapshot()
    })
})
