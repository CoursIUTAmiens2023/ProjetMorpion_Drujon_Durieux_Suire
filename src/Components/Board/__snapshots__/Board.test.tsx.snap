// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should match the snapshot when square are O 1`] = `
<div>
  <div
    class="border-2 border-[#53366c]"
  >
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="0"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="1"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="2"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
    </div>
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="3"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="4"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="5"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
    </div>
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="6"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="7"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
        id="8"
      >
        <span
          class="text-4xl"
        >
          O
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Board should match the snapshot when square are X 1`] = `
<div>
  <div
    class="border-2 border-[#53366c]"
  >
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="0"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="1"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="2"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
    </div>
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="3"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="4"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="5"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
    </div>
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="6"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="7"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
        id="8"
      >
        <span
          class="text-4xl"
        >
          X
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Board should match the snapshot when square are null 1`] = `
<div>
  <div
    class="border-2 border-[#53366c]"
  >
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="0"
      >
        <span
          class="text-4xl"
        />
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="1"
      >
        <span
          class="text-4xl"
        />
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="2"
      >
        <span
          class="text-4xl"
        />
      </button>
    </div>
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="3"
      >
        <span
          class="text-4xl"
        />
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="4"
      >
        <span
          class="text-4xl"
        />
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="5"
      >
        <span
          class="text-4xl"
        />
      </button>
    </div>
    <div
      class="flex"
    >
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="6"
      >
        <span
          class="text-4xl"
        />
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="7"
      >
        <span
          class="text-4xl"
        />
      </button>
      <button
        class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
        id="8"
      >
        <span
          class="text-4xl"
        />
      </button>
    </div>
  </div>
</div>
`;
